pkg/operator/configobservation/auth/auth_metadata.go

```diff
@@ -0,0 +1,112 @@
+package auth
+
+import (
+	"github.com/golang/glog"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/configobservation"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+)
+
+const (
+	targetNamespaceName   = "openshift-kube-apiserver"
+	oauthMetadataFilePath = "/etc/kubernetes/static-pod-resources/configmaps/oauth-metadata/oauthMetadata"
+	configNamespace       = "openshift-config"
+	managedNamespace      = "openshift-config-managed"
+)
+
+// ObserveAuthMetadata fills in authConfig.OauthMetadataFile with the path for a configMap referenced by the authentication
+// config.
+func ObserveAuthMetadata(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	listers := genericListers.(configobservation.ResourceSyncListers)
+	errs := []error{}
+	prevObservedConfig := map[string]interface{}{}
+
+	topLevelMetadataFilePath := []string{"authConfig", "oauthMetadataFile"}
+	currentMetadataFilePath, _, err := unstructured.NestedString(existingConfig, topLevelMetadataFilePath...)
```

Comment:
Use `SetNestedStringSlice` instead

Reply:
> Use `SetNestedStringSlice` instead

that seems odd.  It's a straight string field, not a slice, right?

Reply:
yep, just a string field so using SetNestedStringSlice does not seem right to me.

Reply:
yeah, sorry was looking at two similar stuff at once